datasets/flwr_datasets/federated_dataset.py

```diff
@@ -102,9 +114,18 @@ def __init__(
         # Indicate if the dataset is prepared for `load_partition` or `load_full`
         self._dataset_prepared: bool = False
 
-    def load_partition(self, node_id: int, split: Optional[str] = None) -> Dataset:
+    def load_partition(
+        self,
+        node_id: int,
+        split: Optional[str] = None,
+    ) -> Union[Dataset, List[Dataset], DatasetDict]:
         """Load the partition specified by the idx in the selected split.
 
+        If the partition is divided as specified in `partition_split` then you can
+        return only the specific split by specifying `inner_split` otherwise all split
+        are return. If the `partition_split` was specified as `Dict` give the string
```

Comment:
I feel like `partition_split` and `inner_split` arguments are supposed to be passed to this function but they aren't?

Reply:
Updated.
Initially, I named the `partition_division` as `partition_split` but I thought that I didn't want to keep the split in the name to avoid confusion.
`inner_split` was finally removed it. I thought that the load_partition can be used to already decide which part of the removed data is returned but I thought that the confusion overweight the benefits.